### PR TITLE
Correct bug in read_sub

### LINKED ISSUE #48180: Seems like a bug. Where is the `to` defined?
https://github.com/JuliaLang/julia/blob/708d1bdb36e962b2eb8b2040777f5dea272c40c7/base/iobuffer.jl#L206

Got it from JET.jl
```│┌ @ io.jl:1000 nr = readbytes!(s, b, nb)
││┌ @ stream.jl:889 readbytes!(s, a, Base.Int(nb))
│││┌ @ stream.jl:911 nread = readbytes!(sbuf, a, nb)
││││┌ @ iobuffer.jl:463 Base.read_sub(io, b, 1, nr)
│││││┌ @ iobuffer.jl:206 read(Base.to, T)
││││││ `Base.to` is not defined
│││││└───────────────────
```

### PR DESCRIPTION
**Pull request by gbaraldi**
Should fix #48180

### CODE REVIEW

File: test/iobuffer.jl [hunk 1]
```diff
@@ -348,3 +348,12 @@ end
...
     @test bytesavailable(devnull) == 0
 end
+
+@testset "#48188 read_sub for non Array AbstractArray" begin
+    a = [0,0,0]
+    v = @view a[1:2]
+    io = IOBuffer()
+    write(io,1)
+    seek(io,0)
+    @test Base.read_sub(io,v,1,1) == [1,0]
```
Comment (fredrikekre) [inline test/iobuffer.jl:358]: If there is no surface API that reaches this, perhaps the `else` block should just be removed?

Comment (gbaraldi) [reply test/iobuffer.jl:358]: In fact since it's only used in a single place we might even want to inline it in readbytes!

Comment (vtjnash) [reply test/iobuffer.jl:358]: yeah, we switched to unsafe_read, so go for it

Comment (gbaraldi) [reply test/iobuffer.jl:358]: It makes the code uglier so I guess it's fine. This is good to go IMO

**Pull request merged**